inference-engine/src/vpu/graph_transformer/src/stages/dynamic_shape_resolver.cpp

```diff
@@ -19,6 +19,12 @@ void FrontEnd::parseDSR(const Model& model, const ie::CNNLayerPtr& layer, const
     const auto dataProducerEdge = data->producerEdge();
     VPU_THROW_UNLESS(dataProducerEdge != nullptr, "Parsing layer {} of type {} failed: input with index {} (of name {}) must have a producer",
         layer->name, layer->type, 0, data->name());
+    VPU_THROW_UNLESS(data->consumerEdges().size() == 0,
```

Comment:
This can be true even if data has more than one child. Other consumers may not have been parsed at this time, so this check will pass.

Reply:
Nice catch! Another assertion was added in the last commit. `getInputTo` map of original data is checked there